test.js

```diff
@@ -40,3 +40,14 @@ test('concurrency: 4', async t => {
 test('handles empty iterable', async t => {
 	t.deepEqual(await m([], mapper), []);
 });
+
+test('async with concurrency: 2', async t => {
+	const myInput = [100, 200, 10, 36, 13, 45];
+	const myMapper = value => {
+		return new Promise(resolve => {
+			setTimeout(() => resolve(value), value);
+		});
```

Comment:
@parro-it Would you mind also adding a test with hard-coded timeout where the first element takes longer than the second, so we're 100% sure it supports order?